### PR TITLE
Fix incorrect passenger ordering

### PR DESCRIPTION
**Pull request by Lazerbeak12345**
This actually also fixes a bug where an infinite number of passengers could be mounted to the last passenger seat.

I'll handle the new API changes in a new PR after this pr is done: https://github.com/minetest-mods/lib_mount/pull/4

### CODE REVIEW

File: framework.lua [hunk 1]
```diff
@@ -87,14 +87,12 @@ function vehicle_mash.register_vehicle(name, def)
 						lib_mount.detach(self.passenger3, self.offset)
 					-- if clicker is not passenger
 					else
-						-- attach passengers if possible
-						if lib_mount.passengers[self.passenger] == self.passenger and self.number_of_passengers >= 1 then
+						-- attach a passenger if possible
+						if self.passenger==nil and self.number_of_passengers >= 1 then
```
Comment (Panquesito7) [inline framework.lua:91]: Why `nil` here?

**Review by Panquesito7 — COMMENTED** [review]
A question: does this depend on the API changes PR? https://github.com/minetest-mods/lib_mount/pull/4

**Comment by Lazerbeak12345** [comment]
> A question: does this depend on the API changes PR? [minetest-mods/lib_mount#4](https://github.com/minetest-mods/lib_mount/pull/4)

No, I made sure that it doesn't. As you can see I'm addressing `passenger` `passenger2` and `passenger3` instead of the array `passengers`

File: framework.lua (hunk 1 above)
Comment (Lazerbeak12345) [reply framework.lua:91]: In `lib_mount` it sets each passenger to `nil` when unmounting them. This code is looking for the next empty seat and attaching the passenger to that seat, so the seat must be empty.

**Review by Panquesito7 — APPROVED** [review]
Code looking good, thanks! 🚀 Will test when I get my PC working.

**Comment by Lazerbeak12345** [comment]
Anything I can do to help with that?

**Comment by SmallJoker** [comment]
@Panquesito7  did you fix your PC already?

**Comment by Panquesito7** [comment]
> @Panquesito7 did you fix your PC already?

No, not yet. Now it has other problems, so I can't test it, not even with another disk it turns on (offtopic, though).
I could try testing it on another PC, though. Right now I'm using a laptop but don't wanna add games to this laptop or any other files/stuff. 😅

**Review by Panquesito7 — APPROVED** [review]
Thank you! 🚀

**Pull request merged**